### PR TITLE
feat/tts_mutator

### PR DESCRIPTION
**Pull request by JarbasAl**
add effects to TTS via sox, such as speed and pitch changes

ported from https://github.com/HelloChatterbox/text2speech/blob/master/text2speech/modules/__init__.py#L257

obsoleted by https://github.com/OpenVoiceOS/ovos-plugin-manager/pull/179

### CODE REVIEW

**Review by ChanceNCounter — COMMENTED** [review]
1100 lines is gonna take 5 or 10 passes to review, I suggest divide and conquer

**Comment by JarbasAl** [comment]
it's just wrappers around the sox effects with lots of docstrs, you only need to validate a few of them in practice and confirm kwargs are passed correctly to the subprocess

File: ovos_plugin_manager/templates/tts.py [hunk 1]
```diff
@@ -1059,3 +1075,1044 @@ def get_tts(self, sentence, wav_file, lang=None):
...
+        effect_args = [
+            'treble', '{:f}'.format(gain_db), '{:f}'.format(frequency),
+            '{:f}s'.format(slope)
+        ]
+
+        self.effects.extend(effect_args)
+
+    def tremolo(self, speed=6.0, depth=40.0):
+        """
+        Apply a tremolo (low frequency amplitude modulation) effect to the audio. The tremolo frequency in Hz is giv en by speed, and the depth as a percentage by depth (default 40).
```
Comment (mikejgray) [inline ovos_plugin_manager/templates/tts.py:1383]: Nit: `given` not `giv en`

File: ovos_plugin_manager/templates/tts.py [hunk 2]
```diff
@@ -1059,3 +1075,1044 @@ def get_tts(self, sentence, wav_file, lang=None):
...
+            )
+
+        effect_args = ['speed', '{:f}'.format(factor)]
+
+        self.effects.extend(effect_args)
+
+    def chorus(self, gain_in=0.5, gain_out=0.9, n_voices=3, delays=None,
+               decays=None, speeds=None, depths=None, shapes=None):
+        """
+        Add a chorus effect to the audio. This can makeasingle vocal sound like a chorus, but can also be applied to instrumentation.
```
Comment (mikejgray) [inline ovos_plugin_manager/templates/tts.py:1432]: Nit: `make a single` instead of `makeasingle`

**Review by mikejgray — COMMENTED** [review]
I like this feature in general and it should be a lot of fun for different use cases. However, I'd feel a lot more comfortable approving it if there were some tests. Since we have so many possible interactions with the different sox flags I can imagine a lot of scenarios where they interact poorly.

**Comment by JarbasAl** [comment]
replaced by https://github.com/OpenVoiceOS/ovos-plugin-manager/pull/179

**Pull request closed without merging**